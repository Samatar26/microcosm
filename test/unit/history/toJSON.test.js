import History from '../../../src/history'

describe('History::toJSON', function() {
  const action = n => n

  it('includes the size of the tree', function() {
    const history = new History({ maxHistory: Infinity })

    history.append(action).resolve()
    history.append(action).resolve()

    let json = JSON.parse(JSON.stringify(history))

    expect(json.size).toEqual(3)
  })

  it('includes an id reference to the head', function() {
    const history = new History({ maxHistory: Infinity })

    history.append(action).resolve()

    let json = JSON.parse(JSON.stringify(history))

    expect(json.head).toEqual(history.head.id)
  })

  it('includes an id reference to the root', function() {
    const history = new History({ maxHistory: Infinity })

    history.append(action).resolve()

    let json = JSON.parse(JSON.stringify(history))

    expect(json.root).toEqual(history.root.id)
  })

  it('serializes the tree of actions', function() {
    const history = new History({ maxHistory: Infinity })

    history.append(action).resolve(true)

    expect(history).toMatchSnapshot()
  })
})
