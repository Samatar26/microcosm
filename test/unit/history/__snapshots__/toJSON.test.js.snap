// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History::toJSON serializes the tree of actions 1`] = `
Object {
  "head": "action12",
  "root": "action10",
  "size": 2,
  "tree": Object {
    "children": Array [
      Object {
        "children": Array [],
        "disabled": false,
        "id": "action12",
        "next": null,
        "payload": true,
        "revisions": Array [
          Object {
            "payload": true,
            "status": "resolve",
            "timestamp": 1499687103553,
          },
        ],
        "status": "resolve",
        "type": "action.3",
      },
    ],
    "disabled": false,
    "id": "action10",
    "next": "action12",
    "payload": undefined,
    "revisions": Array [
      Object {
        "payload": undefined,
        "status": "resolve",
        "timestamp": 1499687103553,
      },
    ],
    "status": "resolve",
    "type": "$start.1",
  },
}
`;
